echo "Enter first read R1";
read filename1;
echo "Enter second read R2";
read filename2;
echo "Pre-processing FASTQ files for GATK variant calling";
java -Xmx8G -jar /home/ngs/Downloads/picard-tools-1.141/picard.jar FastqToSam FASTQ=$filename1 FASTQ2=$filename2 OUTPUT=sample_fastq2sam.bam READ_GROUP_NAME=H0164.2 SAMPLE_NAME=name LIBRARY_NAME=Solexa-272222 PLATFORM_UNIT=H0164ALXX140820.2 PLATFORM=illumina SEQUENCING_CENTER=BI RUN_DATE=2018-06-04T00:00:0400;
java -Xmx8G -jar /home/ngs/Downloads/picard-tools-1.141/picard.jar MarkIlluminaAdapters I=sample_fastq2sam.bam o=sample_markilluminaadapters.bam M=sample_markilluminaadapters_metrics.txt;
java -Xmx8G -jar /home/ngs/Downloads/picard-tools-1.141/picard.jar SamToFastq I=sample_markilluminaadapters.bam FASTQ=sample_markilluminaadapters_R1.fastq SECOND_END_FASTQ=sample_markilluminaadapters_R2.fastq CLIPPING_ATTRIBUTE=XT CLIPPING_ACTION=2 NON_PF=true;
bwa mem -M -t 16 /mnt/NGS/Human_Exome_hg19/hg19 sample_markilluminaadapters_R1.fastq sample_markilluminaadapters_R2.fastq > sample_bwa_mem.sam;
java -Xmx8G -jar /home/ngs/Downloads/picard-tools-1.141/picard.jar MergeBamAlignment R=/mnt/NGS/Human_Exome_hg19/hg19.fa UNMAPPED_BAM=sample_fastq2sam.bam ALIGNED_BAM=sample_bwa_mem.sam o=sample_mergebamalignment.bam CREATE_INDEX=true ADD_MATE_CIGAR=true CLIP_ADAPTERS=false CLIP_OVERLAPPING_READS=true INCLUDE_SECONDARY_ALIGNMENTS=true MAX_INSERTIONS_OR_DELETIONS=-1 PRIMARY_ALIGNMENT_STRATEGY=MostDistant ATTRIBUTES_TO_RETAIN=XS;
java -Xmx8g -jar /home/ngs/Downloads/picard-tools-1.141/picard.jar MarkDuplicates INPUT=sample_mergebamalignment.bam OUTPUT=sample_markduplicates.bam METRICS_FILE=sample_markduplicates_metrics.txt OPTICAL_DUPLICATE_PIXEL_DISTANCE=2500 CREATE_INDEX=true;
echo "Executing GATK commands";
java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /mnt/NGS/Human_Exome_hg19/hg19.fa -L truseq-exome-targeted-regions-manifest-v1-2_CCMB.bed -known /home/ngs/Downloads/GenomeAnalysisTK-3.5/GATK_bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19_sorted.vcf -I sample_markduplicates.bam -o realignertargetcreator.intervals -ip 100;
java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /mnt/NGS/Human_Exome_hg19/hg19.fa -targetIntervals realignertargetcreator.intervals -known /home/ngs/Downloads/GenomeAnalysisTK-3.5/GATK_bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19_sorted.vcf -I sample_markduplicates.bam -o indelrealigner.bam;
java -Xmx8G -jar /home/ngs/Downloads/picard-tools-1.141/picard.jar ValidateSamFile I=indelrealigner.bam MODE=SUMMARY o=validate_report.txt;
## to print certain types of reads that cause problems for downstream GATK tools
## java  -jar /path/ GenomeAnalysisTK.jar -T PrintReads -R /mnt/NGS/Human_Exome_hg19/hg19.fa (desired genomic interval chr:start-end) -I indelrealigner.bam -o output.bam;
java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /mnt/NGS/Human_Exome_hg19/hg19.fa -I indelrealigner.bam -L truseq-exome-targeted-regions-manifest-v1-2_CCMB.bed -ip 100 -knownSites /home/ngs/Downloads/GenomeAnalysisTK-3.5/GATK_bundle/hg19/dbsnp_138.hg19_sorted.vcf -knownSites /home/ngs/Downloads/GenomeAnalysisTK-3.5/GATK_bundle/hg19/Mills_and_1000G_gold_standard.indels.hg19_sorted.vcf -o recal_data.table;
##java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /mnt/NGS/Human_Exome_hg19/hg19.fa -I indelrealigner.bam -knownSites /home/ngs/Downloads/GenomeAnalysisTK-3.5/GATK_bundle/hg19/dbsnp_138.hg19_sorted.vcf -BQSR recal_data.table -o post_recal_data.table;
##java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /mnt/NGS/Human_Exome_hg19/hg19.fa -before recal_data.table -after post_recal_data.table -plots recalibration_plots.pdf;
java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /mnt/NGS/Human_Exome_hg19/hg19.fa -I indelrealigner.bam -BQSR recal_data.table -o smp.bam;
java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /mnt/NGS/Human_Exome_hg19/hg19.fa -I Arshiya.bam -L truseq-exome-targeted-regions-manifest-v1-2_CCMB.bed -ip 100 --genotyping_mode DISCOVERY -stand_emit_conf 10 -stand_call_conf 30 -o raw_variants.vcf;
echo "select variants";
java -Xmx8G -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /mnt/NGS/Human_Exome_hg19/hg19.fa -V raw_variants.vcf -selectType SNP -o raw_snps.vcf;
java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /mnt/NGS/Human_Exome_hg19/hg19.fa -V raw_snps.vcf --filterExpression "QD < 2.0" --filterName "QualDepth" --filterExpression "FS > 60.0" --filterName "FisherStrand" --filterExpression "MQ < 40.0" --filterName "RMSMapping" --filterExpression "MQRankSum < -12.5" --filterName "MappingQualityRankSumTest" --filterExpression "ReadPosRankSum < 8.0" --filterName "ReadPosRankSumTest" -o filtered_snps.vcf;
java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /mnt/NGS/Human_Exome_hg19/hg19.fa -V raw_variants.vcf -selectType INDEL -o raw_indels.vcf;
java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /mnt/NGS/Human_Exome_hg19/hg19.fa -V raw_indels.vcf --filterExpression "QD < 2.0" --filterName "QualByDepth" --filterExpression "FS > 200.0" --filterName "FisherStrand" --filterExpression "ReadPosRankSum < -20.0" --filterName "ReadPosRankSum" -o filtered_indels.vcf;
java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T CombineVariants -R /mnt/NGS/Human_Exome_hg19/hg19.fa -V filtered_snps.vcf -V filtered_indels.vcf -genotypeMergeOptions UNSORTED -o sample_hardfiltered.vcf;
##java -jar GenomeAnalysisTK.jar -T ApplyRecalibration -R /hg19.fa -input /path/sample_unfiltered.vcf -mode SNP -ts_filter /path/recalibrate_SNP.recal -tranchesFile /path/recalibrate_SNP.tranches -O /path/recalibrated_snps_indels.vcf;
##java -jar /path/GenomeAnalysisTK.jar -T VariantRecalibrator -R /hg19.fa -input /sample_recalibrated_snps_raw_indels.vcf -resource:mills, known=true, training=true, truth=true, prior=12.0 /path/Mills_and_1000G_gold_standard.indels.hg19_sorted.vcf -an QD -an FS -an SOR -an MQRankSum -mode INDEL -tranche 100.0 -tranche 99.9 -tranche 90.0 --maxGaussians 4 -recalFile /path/recalibrate_INDEL.recal -tranchesFile /path/recalibrate_INDEL.tranches -rscriptFile /path/recalibrate_INDEL_plots.R;
##java -jar /path/GenomeAnalysisTK.jar -T ApplyRecalibration -R /path/hg19.fa -input /recalibrated_snps_raw_indels.vcf -mode INDEL --ts_filter_level 99.0 -recalFile /path/recalibrate_INDEL.recal -tranchesFile /path/recalibrate_INDEL.tranches -O sample_recalibrated_variant.vcf;
echo "going to perform validatevariants";
##java -jar /home/ngs/Downloads/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T ValidateVariants -R /mnt/NGS/Human_Exome_hg19/hg19.fa -V sample_hardfiltered --dbsnp /home/ngs/Downloads/GenomeAnalysisTK-3.5/GATK_bundle/hg19/dbsnp_138.hg19_sorted.vcf -o validated-variants;
echo "making annovar";
perl /mnt/Diag/Annovar/annovar/convert2annovar.pl -format vcf4 sample_hardfiltered.vcf -outfile smp.avinput -includeinfo -withzyg;
perl /mnt/Diag/Annovar/annovar/table_annovar.pl Arshiya.avinput /mnt/Diag/Annovar/humandb_hg19/ -buildver hg19 -out smp_anno -remove -protocol refGene,phastConsElements46way,1000g2015aug_all,esp6500siv2_all,gnomad211_exome,gnomad211_genome,cg69,gme,Generic_All-Indian_Variants,Generic_GAsP_SNVs_July20,snp138NonFlagged,avsnp150,clinvar_20200316,ljb26_sift,ljb26_pp2hdiv,ljb26_pp2hvar,ljb26_mt,caddindel,cadd13,mcap,dbscsnv11,intervar_20180118 -operation g,r,f,f,f,f,f,f,f,f,f,f,f,f,f,f,f,f,f,f,f,f -genericdbfile hg19_Generic_All-Indian_Variants.txt -genericdbfile hg19_Generic_GAsP_SNVs_July20.txt --nastring NA -otherinfo;
echo adding OMIM genes and phenotypes;
perl /mnt/Diag/WGS-ikrormi/OMIM/OMIM_Gene.pl Casimir_GAsP_anno.hg19_multianno.txt;
/mnt/Diag/WGS-ikrormi/OMIM/OMIM_phenotype.pl OMIM-gene.txt;
#Filter population allele at MAF <0.01
echo filtering population databases;
perl /mnt/Diag/WGS-ikrormi/OMIM/filter.pl OMIM-pheno.txt;
